ignite/internal/tools/gen-mig-diffs/cmd/root.go

```diff
@@ -0,0 +1,187 @@
+			if err != nil {
+				return err
+			}
+			defer sFrom.Cleanup()
+
+			if err := sFrom.Run(cmd.Context()); err != nil {
+				return err
+			}
+			session.StopSpinner()
+			session.EventBus().SendInfo(fmt.Sprintf("Scaffolded code for %s at %s", igniteRepo.From.Original(), sFrom.Output))
+
+			session.StartSpinner(fmt.Sprintf("Running scaffold commands for %s...", igniteRepo.To.Original()))
+			sTo, err := scaffold.New(toBin, igniteRepo.To, scaffoldOptions...)
+			if err != nil {
+				return err
+			}
+			defer sTo.Cleanup()
+
+			if err := sTo.Run(cmd.Context()); err != nil {
+				return err
+			}
+			session.StopSpinner()
+			session.EventBus().SendInfo(fmt.Sprintf("Scaffolded code for %s at %s", igniteRepo.To.Original(), sTo.Output))
+
+			// Calculate and save the diffs from the scaffolded code.
+			session.StartSpinner("Calculating diff...")
+			diffs, err := diff.CalculateDiffs(sFrom.Output, sTo.Output)
+			if err != nil {
+				return errors.Wrap(err, "failed to calculate diff")
+			}
+
+			formatedDiffs, err := diff.FormatDiffs(diffs)
+			if err != nil {
+				return errors.Wrap(err, "failed to save diff map")
+			}
+			session.StopSpinner()
+			session.EventBus().SendInfo("Diff calculated successfully")
+
+			output, err = filepath.Abs(output)
+			if err != nil {
+				return errors.Wrap(err, "failed to find the abs path")
+			}
+
+			// Generate the docs file.
+			g, err := doc.NewGenerator(doc.Options{
+				Path:        output,
+				FromVersion: igniteRepo.From,
+				ToVersion:   igniteRepo.To,
+				Diffs:       string(formatedDiffs),
+				Description: releaseDescription,
+			})
+			if err != nil {
+				return errors.Wrap(err, "failed to create the doc generator object")
+			}
+
+			sm, err := xgenny.NewRunner(cmd.Context(), output).RunAndApply(g)
+			if err != nil {
+				return err
+			}
+
+			if len(sm.CreatedFiles()) == 0 {
+				return errors.Errorf("migration doc not created at %s", output)
+			}
+			session.EventBus().SendInfo(
+				fmt.Sprintf("Migration doc generated successfully at %s", sm.CreatedFiles()[0]),
+			)
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringP(flagFrom, "f", "", "Version of ignite or path to ignite source code to generate the diff from")
+	cmd.Flags().StringP(flagTo, "t", "", "Version of ignite or path to ignite source code to generate the diff to")
+	cmd.Flags().StringP(flagOutput, "o", defaultDocPath, "Output directory to save the migration document")
+	cmd.Flags().StringP(flagSource, "s", "", "Path to ignite source code repository. Set the source automatically set the cleanup to false")
+	cmd.Flags().String(flagRepoURL, "", "Git URL for the Ignite repository")
+	cmd.Flags().String(flagRepoOutput, "", "Output path to clone the ignite repository")
+	cmd.Flags().Bool(flagRepoCleanup, true, "Cleanup the repository path after use")
+	cmd.Flags().String(flagScaffoldOutput, "", "Output path to clone the ignite repository")
+	cmd.Flags().String(flagScaffoldCache, "", "Output path to clone the ignite repository")
```

Comment:
I think the help text for this one "got left behind":

```suggestion
	cmd.Flags().String(flagScaffoldCache, "", "Path to Ignite cache directory")
```